src/main/java/com/teragrep/cfe18/handlers/CaptureGroupController.java

```diff
@@ -144,7 +143,7 @@ public List<CaptureGroup> getAllCaptureGroup(@RequestParam(required = false) Int
     public ResponseEntity<String> newCaptureGroup(@RequestBody CaptureGroup newCaptureGroup) {
         LOGGER.info("About to insert <[{}]>",newCaptureGroup);
         JSONObject jsonErr = new JSONObject();
-        jsonErr.put("id", 0);
+        jsonErr.put("id", newCaptureGroup.getCapture_definition_id());
```

Comment:
please create an issue about refactoring this. creating a new capture group and putting captures into it are separate responsibilities.